.github/workflows/hubble-cli.yaml

```diff
@@ -0,0 +1,147 @@
+name: Hubble CLI tests
+
+on:
+  pull_request: {}
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number }}
+  cancel-in-progress: true
+
+env:
+  # renovate: datasource=github-releases depName=helm/helm
+  HELM_VERSION: v3.14.3
+  # renovate: datasource=github-releases depName=kubernetes-sigs/kind
+  KIND_VERSION: v0.22.0
+  KIND_CONFIG: .github/kind-config-single-node.yaml
+  # renovate: datasource=github-releases depName=cilium/cilium
+  CILIUM_VERSION: v1.15.3
+
+jobs:
+  check_changes:
+    name: Deduce required tests from code changes
+    runs-on: ubuntu-22.04
+    outputs:
+      tested: ${{ steps.tested-tree.outputs.src }}
+    steps:
+      - name: Check code changes
+        uses: dorny/paths-filter@de90cc6fb38fc0963ad72b210f1f284cd68cea36 # v3.0.2
+        id: tested-tree
+        with:
+          filters: |
+            src:
+              - '!(test|Documentation)/**'
+
+  build-hubble-cli-release-binaries:
+    name: Build Hubble CLI release binaries
+    needs: check_changes
+    if: ${{ needs.check_changes.outputs.tested == 'true' }}
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout the repository
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+      - name: Setup go
+        uses: actions/setup-go@0c52d547c9bc32b1aa3301fd7a9cb496313a4491 # v5.0.0
+        with:
+          # renovate: datasource=golang-version depName=go
+          go-version: 1.22.2
+      - name: Build hubble CLI release binaries
+        run: |
+          make -C hubble local-release
+
+  integration-test:
+    name: Run hubble CLI integration tests
+    needs: check_changes
+    if: ${{ needs.check_changes.outputs.tested == 'true' }}
+    runs-on: ubuntu-22.04
+    timeout-minutes: 20
+    steps:
+      - name: Checkout the repository
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+      - name: Setup go
+        uses: actions/setup-go@0c52d547c9bc32b1aa3301fd7a9cb496313a4491 # v5.0.0
+        with:
+          # renovate: datasource=golang-version depName=go
+          go-version: 1.22.2
+      - name: Build Hubble CLI
+        run: make -C hubble
+      - name: Set up Helm
+        uses: azure/setup-helm@5119fcb9089d432beecbf79bb2c7915207344b78 # renovate: tag=v3.5
+        with:
+          version: ${{ env.HELM_VERSION }}
+      - name: Create kind cluster
+        uses: helm/kind-action@99576bfa6ddf9a8e612d83b513da5a75875caced # v1.9.0
+        with:
+          version: ${{ env.KIND_VERSION }}
+          config: ${{ env.KIND_CONFIG }}
+          cluster_name: kind
+      - name: Deploy Cilium
+        run: |
+          # Deploy cilium with KPR
+          MASTER_IP="$(docker inspect kind-control-plane | jq '.[0].NetworkSettings.Networks.kind.IPAddress' -r)"
+          helm repo add cilium https://helm.cilium.io
+          helm repo update
+          helm install cilium cilium/cilium \
```

Comment:
This is installing the latest stable version of Cilium - which means it doesn't test any Hubble API additions present on the PR branch or on `main`, which is one of the reasons we want to do the merge in the first place.

Since I believe we will want to change this workflow significantly, I wonder if it even makes sense to add this version of the workflow to cilium/cilium, since I expect major rewrites are necessary to make it fit the style of workflows we have in cilium/cilium (e.g. using cilium-cli sysdump rather than our own "Gathering information about KIND cluster" step below).

I would vote for not adding this workflow in it's current state and instead have a follow-up task to have this workflow redesigned from scratch (or have it become part to one of the existing CI tests, e.g. ConformanceKind)

Reply:
sounds good, let me delete the integration test. i'll keep the `build-hubble-cli-release-binaries` job.

i'll update the CFP and add a task for creating new integration tests ✨